### ISSUE TITLE
Swift version specific rules

### ISSUE THREAD

**Issue by jonah-williams** [realm/SwiftLint#700]
Do you all have thoughts about how to manage sets of rules which are Swift 2.3 or Swift 3.0 specific?

Given Xcode 8 supporting both Swift versions how might we allow developers using SwiftLint to specify which set of rules should apply to a specific project? We could maintain rules with different names and include and exclude them individually but that's not a great interface to work with.

Should we specify the swift version as part of of our `.swiftlint.yml`? Could we resolve this at the same time as #221 by specifying a version of the ruleset to apply?

**Comment by jpsim** [comment]
SwiftLint works against any Swift version after 1.0, and will resolve the Swift version to use in the following order:

* `$XCODE_DEFAULT_TOOLCHAIN_OVERRIDE`
* `$TOOLCHAIN_DIR`
* `xcrun -find swift`
* `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain`
* `/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain`
* `~/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain`
* `~/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain`

That being said, I'm not aware of any rule that _semantically_ should behave differently across Swift versions other than the change that came in #660.

If you can think of other rules that might need to behave differently with Swift 3, please let us know!

**Comment by andreyz** [comment]
@jpsim rule `conditional_binding_cascade` needs changing for Swift 3, since it triggers a lot of false positive for completely legal code.

```swift
guard let item = item, let delegate = delegate, item.isEmpty else { return false }
```

_Note `let` for every binding and comma instead of Swift 2+ `where`._

**Comment by jpsim** [comment]
As of #1081, we can detect the current Swift version and adjust rule behavior based on that.
